### PR TITLE
[Customer][Payment] Disabled payment methods show in Customer Dashboard

### PR DESCRIPTION
**Pull request by vasilii-b**
### Description (*)
This PR aims to: 
- Hide Billing Agreements link from customer account if Paypal payment method is disabled or the Billing Agreements Signup method is set to Never
- Hide Stored Payment Methods link from customer account if Braintree payment method is disabled

### Fixed Issues (if relevant)
1. magento/magento2#6659: Disabled payment methods show in Customer Dashboard

### Manual testing scenarios (*)
Provided in issue description.

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

**Comment by magento-engcom-team** [comment]
Hi @vasilii-b. Thank you for your contribution 
Here is some useful tips how you can test your changes using Magento test environment.
Add the comment under your pull request to deploy test or vanilla Magento instance:
- `@magento-engcom-team give me test instance` - deploy test instance based on PR changes
- `@magento-engcom-team give me 2.3-develop instance` - deploy vanilla Magento instance

For more details, please, review the [Magento Contributor Assistant](https://devdocs.magento.com/guides/v2.2/contributor-guide/contributing.html#contributor-assist) documentation

File: app/code/Magento/Paypal/Block/Customer/Account/BillingAgreementsLink.php [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+    /**
+     * @return string
+     */
+    protected function _toHtml()
+    {
+        $moduleEnabled = $this->_scopeConfig->getValue(self::XML_PATH_PAYPAL_PAYMENT_ENABLED);
+        $baSignupType = $this->_scopeConfig->getValue(self::XML_PATH_BILLING_AGREEMENT_SIGNUP);
+
+        if (
```
Comment (rogyar) [inline app/code/Magento/Paypal/Block/Customer/Account/BillingAgreementsLink.php]: Could make this condition in a single line?

**Review by rogyar — CHANGES_REQUESTED** [review]

File: app/code/Magento/Paypal/Block/Customer/Account/BillingAgreementsLink.php (hunk 1 above)
Comment (vasilii-b) [reply app/code/Magento/Paypal/Block/Customer/Account/BillingAgreementsLink.php]: @rogyar I've fixed the issue. Thank you!

**Comment by rogyar** [comment]
@vasilii-b please, check the failing tests

**Comment by sidolov** [comment]
@vasilii-b , I am closing this PR now due to inactivity.
Please reopen and update if you wish to continue.
Thank you for the collaboration!

File: app/code/Magento/Paypal/Block/Customer/Account/BillingAgreementsLinkInterface.php [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+namespace Magento\Paypal\Block\Customer\Account;
+
+use Magento\Customer\Block\Account\SortLinkInterface;
+
+/**
+ * BillingAgreementsLinkInterface
+ * @api
+ */
+interface BillingAgreementsLinkInterface extends SortLinkInterface
```
Comment (rogyar) [inline app/code/Magento/Paypal/Block/Customer/Account/BillingAgreementsLinkInterface.php:15]: I'm not sure it's a good idea to introduce a new interface just for storing config paths. Maybe we should move it to a separate config model or to the block class itself. 

@sidolov could you share your thoughts on this matter, please? 
Thank you

**Review by rogyar — CHANGES_REQUESTED** [review]

File: app/code/Magento/Paypal/Block/Customer/Account/BillingAgreementsLinkInterface.php (hunk 2 above)
Comment (sidolov) [reply app/code/Magento/Paypal/Block/Customer/Account/BillingAgreementsLinkInterface.php:15]: @rogyar totally agree with you. If constants used in one model only, we should define them on the particular model as private.

**Comment by sidolov** [comment]
@vasilii-b , I am closing this PR now due to inactivity.
Please reopen and update if you wish to continue.
Thank you for the collaboration!

**Pull request closed without merging**